.github/workflows/ssh-azure-tee-build.yaml

```diff
@@ -0,0 +1,64 @@
+name: sgx-build-azure-ssh
+
+on:
+  push:
+    branches: [ "quote-presentation" ]
+  pull_request:
+    branches: [ "quote-presentation" ]
+    
+# perms needed for attestation workflow
+permissions:
+  id-token: write
+  attestations: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: tee
+    steps:
+    - name: azure-tee-build-${{ github.sha }}
+      uses: appleboy/ssh-action@v1.0.3
+      env:
+         # using personal access token until devops links acct
+         PAT_TOKEN: ${{ secrets.PAT_SET_ENV }}
+      with:
+         host: ${{ secrets.AZURE_TEE_BUILD_HOST }}
+         username: ${{ secrets.AZURE_TEE_BUILD_USERNAME }}
+         key: ${{ secrets.AZURE_TEE_BUILD_KEY }}
+         port: ${{ secrets.SSH_PORT }}
+         command_timeout: 10m
+         allenvs: true
+         envs: PAT_TOKEN
+         script: |
+            cd /tmp 
+            source $HOME/.cargo/env
+            rm -rf ${{ github.sha }}
+            git clone https://github.com/tlsnotary/tlsn ${{ github.sha }}
+            cd ${{ github.sha }}
+            git checkout ${{ github.sha }}
+            cd crates/notary/server/config/gramine
+            random_port=$(shuf -i 1024-65535 -n 1)
+            sed -i '/port*: *7047/s/7047/'$random_port'/' ../config.yaml
+            make clean
+            SGX=1 make start-gramine-server &
+            notarypid=$!
+            win=0
+            SECONDS=0; set -o pipefail; while [ $SECONDS -lt 300 ]; do echo -e "GET /info HTTP/1.1\r\nHost: localhost\r\nConnection: Close\r\n\r\n" | openssl s_client -quiet localhost:$random_port 2>&1 | tee s.log; if [ $? -eq 0 ]; then win=1; break; fi; sleep 2; done
+            if [ $win -eq 0 ]; then echo "Build Failed"; kill -SIGTERM $notarypid; exit 1; fi
+            cat s.log | grep '^{' |  jq '. | tostring' > quotejson
+            mapfile quote < quotejson
+            curl -L -X PATCH -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $PAT_TOKEN" -H "Connection: close" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/$GITHUB_REPOSITORY/environments/TEE/variables/AZURE_TEE_BUILD_PORT -d "{\"name\":\"AZURE_TEE_BUILD_PORT\",\"value\":$quote}"
+            kill -SIGTERM $notarypid
+            sleep 5
+            exit 0
```

Comment:
I'd prefer this to be a real script (e.g. in `cd-scripts`), so the same code can be used for development and ci.